ARSIS-Telemetry-GroundControl/telemetry-api/app/routers/location.py

```diff
@@ -15,49 +18,53 @@ class LocationLLAH(BaseModel):
     heading: float
 
 
-# def create_random_location(
-#     latitude_min=-90,
-#     latitude_max=90,
-#     longitude_min=-180,
-#     longitude_max=180,
-#     altitude_min=420,
-#     altitude_max=8848,
-#     heading_min=0,
-#     heading_max=360,
-# ):
-#     latitude = round(random.uniform(latitude_min, latitude_max), 6)
-#     longitude = round(random.uniform(longitude_min, longitude_max), 6)
-#     altitude = random.randint(altitude_min, altitude_max)
-#     heading = random.randint(heading_min, heading_max)
-#     return (latitude, longitude, altitude, heading)
-
-
-# def create_random_string(characters=string.ascii_lowercase):
-#     return "".join(random.choice(characters) for _ in range(6))
+keys = [
+    "id",
+    "longitude",
+    "latitude",
+    "altitude",
+    "heading",
+    "createdAt",
+    "updatedAt",
+]
 
 
 @router.get("/")
 async def location(request: Request):
-    all_users = request.app.user_cache.get_all()
-    users_list = [{**{"user": k}, **v["location"]} for k, v in all_users.items()]
-    data = {"users": users_list}
+    with connection.cursor() as db:
+        db.execute("SELECT * FROM locations;")
+        result = db.fetchall()
+        response = []
+        for row in result:
+            response.append({i: j for i, j in zip(keys, row[1:])})
+        return {"users": response}
 
-    return data
 
-
-@router.get("/{user}")
+@router.get("/{user_id}")
 async def user_location(req: Request, res: Response, user_id: int):
+    with connection.cursor() as db:
+        query = f"SELECT * FROM locations WHERE id = {user_id} ORDER BY createdat DESC LIMIT 1;"
+        db.execute(query)
+        row = db.fetchone()
+        return {i: j for i, j in zip(keys, row[1:])}
     user_info = req.app.user_cache.get(user_id)
     if not user_info:
         res.status_code = status.HTTP_404_NOT_FOUND
         return {"error": f"User with id: {user_id} not found"}
     return user_info["location"]
```

Comment:
do we want to remove lines 50-54? it uses the cache (which i think isnt being updated) and it wont ever be called right? because of the previous return...

Reply:
Yes, feel free to leave this as a not aproval thing. this is what happens when you watch a movie while coding